### PR TITLE
Add EXPLAIN and EXPLAIN ANALYZE support for Postgres.

### PR DESCRIPTION
**Pull request by m4dc4p**
* Added `explain` and `explainAnalyze` extension methods to Query/Query0
when postgres implicits are imported.
* Updated docs with small example.

(follow up from #920).

### CODE REVIEW

File: modules/postgres/src/main/scala/doobie/postgres/syntax/syntax.scala [hunk 1]
```diff
@@ -614,4 +618,73 @@ trait ToPostgresMonadErrorOps {
...
+   * Construct a program in
+   * `[[doobie.free.connection.ConnectionIO ConnectionIO]]` which returns the server's query plan
+   * for the query (i.e., `EXPLAIN` output). The query is not actually executed.
+   */
+  def explain: ConnectionIO[List[String]] =
+    self.inspect { (sql, prepare) =>
+      HC.prepareStatement(s"EXPLAIN $sql")(for {
+        _ <- prepare
+        result <- HPS.executeQuery(HRS.build[List, String])
+      } yield result )
```
Comment (tpolecat) [inline modules/postgres/src/main/scala/doobie/postgres/syntax/syntax.scala]: Can you simplify the `for` comp to `prepare *>  HPS.executeQuery(HRS.build[List, String]` here and in the other instances?

File: modules/docs/src/main/tut/docs/15-Extensions-PostgreSQL.md [hunk 2]
```diff
@@ -301,3 +301,33 @@ We can run it thus, yielding the number of affected rows.
 ### Fastpath
 
 **doobie** provides an algebra and free monad for constructing programs that use the `FastPathAPI` provided by the PostgreSQL JDBC driver, however this API is mostly deprecated in favor of server-side statements (see above). And in any case I can't find an example of how you would use it from Java so I don't have an example here. But if you're using it let me know and we can figure it out.
+
+### EXPLAIN/EXPLAIN ANALYZE
+
+The PostgreSQL server can provide an analysis of any query, using the `EXPLAIN` keyword. **doobie** can run `EXPLAIN`
+on any `Query0` or `Query` object, as long as `doobie.postgres` and `doobie.postgres.implicits._` have been imported.
```
Comment (tpolecat) [inline modules/docs/src/main/tut/docs/15-Extensions-PostgreSQL.md]: `doobie.postgres._`

File: modules/postgres/src/main/scala/doobie/postgres/syntax/syntax.scala (hunk 1 above)
Comment (m4dc4p) [reply modules/postgres/src/main/scala/doobie/postgres/syntax/syntax.scala]: The `*>` method is unfamiliar to me. When I write `HC.prepareStatement(s"EXPLAIN $sql")(prepare *> HPS.executeQuery(HRS.build[List, String]))`, I get a compile error:

```
[error] \doobie\modules\postgres\src\main\scala\doobie\postgres\syntax\syntax.scala:638:52: value *> is not a member of doobie.PreparedStatementIO[Unit]
[error]       HC.prepareStatement(s"EXPLAIN $sql")(prepare *> HPS.executeQuery(HRS.build[List, String]))
```

Which import do I need to make it work?

Comment (tpolecat) [reply modules/postgres/src/main/scala/doobie/postgres/syntax/syntax.scala]: `cats.implicits._` and possibly `doobie.implicits._`

Comment (jbaileyashe) [reply modules/postgres/src/main/scala/doobie/postgres/syntax/syntax.scala]: Thank you! Changes pushed :)

**Pull request merged**